### PR TITLE
Test the federation `/invite` API

### PR DESCRIPTION
**Pull request by leonerd**

### CODE REVIEW

File: tests/50federation/35room-invite.pl [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            assert_eq( $body->{origin}, $user->http->server_name,
+               'event origin' );
+            assert_eq( $body->{room_id}, $room_id,
+               'event room_id' );
+            assert_eq( $body->{sender}, $user->user_id,
+               'event sender' );
+
+            assert_json_keys( $body, qw( content state_key prev_state ));
+
+            assert_deeply_eq( $body->{content}, { membership => "invite" },
```
Comment (illicitonion) [inline tests/50federation/35room-invite.pl]: This should be a subset, not equal, check. The APIs are meant to be extensible.

File: tests/50federation/35room-invite.pl [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            return $event->{type} eq "m.room.member" &&
+                   $event->{room_id} eq $room_id;
+            }
+         )->then( sub {
+            my ( $event ) = @_;
+            log_if_fail "Invitation event", $event;
+
+            assert_eq( $event->{state_key}, $user->user_id,
+               'event state_key' );
+            assert_deeply_eq( $event->{content}, { membership => "invite" },
```
Comment (illicitonion) [inline tests/50federation/35room-invite.pl]: This should be a subset, not equal, check. The APIs are meant to be extensible.

File: tests/50federation/35room-invite.pl [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            assert_deeply_eq( $body->{content}, { membership => "invite" },
+               'event content' );
+            assert_eq( $body->{state_key}, $invitee_id,
+               'event state_key' );
+
+            $inbound_server->datastore->sign_event( $body );
+
+            $req->respond_json(
+               # SYN-490
+               [ 200 => { event => $body } ]
```
Comment (illicitonion) [inline tests/50federation/35room-invite.pl]: , rather than => ?

File: tests/50federation/35room-invite.pl (hunk 1 above)
Comment (leonerd) [inline tests/50federation/35room-invite.pl]: Done 6772712

File: tests/50federation/35room-invite.pl (hunk 2 above)
Comment (leonerd) [inline tests/50federation/35room-invite.pl]: Done 6772712

File: tests/50federation/35room-invite.pl (hunk 3 above)
Comment (leonerd) [inline tests/50federation/35room-invite.pl]: Eh; it doesn't actually matter as they do the same thing here. But yes for consistency I guess it should be `,`.  Done c3f2eb1

**Comment by illicitonion** [comment]
LGTM

**Pull request merged**